typify-impl/src/convert.rs

```diff
@@ -1132,6 +1132,51 @@ impl TypeSpace {
         Ok((TypeEntryDetails::Unit.into(), metadata))
     }
 
+    /// Determine whether a schema's property name validation constraints can be handled
+    fn can_handle_pattern_properties(validation: &ObjectValidation) -> bool {
+        if !validation.required.is_empty() {
+            return false;
+        }
+
+        if !validation.properties.is_empty() {
+            return false;
+        }
+
+        let all_pattern_property_schemas_same = validation
+            .pattern_properties
+            .values()
+            .next()
+            .map(|first| {
+                validation
+                    .pattern_properties
+                    .values()
+                    .all(|schema| schema == first)
+            })
+            .unwrap_or(true);
+
+        // Ensure we have at least one pattern property and all pattern property
+        // schemas are the same
+        if validation.pattern_properties.is_empty() || !all_pattern_property_schemas_same {
+            return false;
+        }
+
+        // Ensure any additional properties are just a false schema
+        if validation.additional_properties.as_ref().map(AsRef::as_ref)
+            != Some(&Schema::Bool(false))
+            && validation.additional_properties.is_some()
```

Comment:
Can this be:

```suggestion
        if validation.additional_properties.as_ref().map(AsRef::as_ref)
            == Some(&Schema::Bool(true))
```

Reply:
I don't think so, since we're trying to only allow `Some(Schema::Bool(false))` or `None` here.

Reply:
Isn't `== true` equivalent to `!= false && Some`?

Reply:
Ah. I misread the diff -- didn't realize line 1162 is unchanged here :P my bad!

Reply:
Maybe the latest change clears it up? I think Mr. De Morgan is failing me (~(A|B) here != ~A & ~B) because the second check is doing extra work (e.g. if additional properties is an object, that's also not allowed -- must be None or false).

Reply:
Ok I think this is fixed. Shouldn't have been this hard, I'm blaming Monday :)